### ISSUE TITLE
Unable to retrieve the mime_type for PHAR files

### ISSUE THREAD

**Issue by jasonmccreary** [thephpleague/flysystem#1448]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | n/a
| Version     | 3.0.17

#### Summary
After upgrading from 3.0.14 to 3.017, I started getting the following error when downloading `.phar` files:

> League\Flysystem\UnableToRetrieveMetadata: Unable to retrieve the mime_type for file at location...

#### How to reproduce
This was taken from a Laravel application, so the snippet may not be relevant:

```php
Storage::disk('shift')->download('shift-' . $sku . '.phar', 'shift');
```

Here's the relevant trace:

> League\Flysystem\UnableToRetrieveMetadata: Unable to retrieve the mime_type for file at location: shift.phar. 
50 /vendor/league/flysystem/src/UnableToRetrieveMetadata.php(49): League\Flysystem\UnableToRetrieveMetadata::create
49 /vendor/league/flysystem/src/UnableToRetrieveMetadata.php(44): League\Flysystem\UnableToRetrieveMetadata::mimeType
48 /vendor/league/flysystem/src/Local/LocalFilesystemAdapter.php(378): League\Flysystem\Local\LocalFilesystemAdapter::mimeType
47 /vendor/league/flysystem/src/Filesystem.php(137): League\Flysystem\Filesystem::mimeType
46 /vendor/laravel/framework/src/Illuminate/Filesystem/FilesystemAdapter.php(551): Illuminate\Filesystem\FilesystemAdapter::mimeType

**Comment by AiAe** [comment]
I have the same exact issue, but with custom file name. 
Why is such check pushed in a patch update if it breaks already existing code?

**Comment by frankdejonge** [comment]
Hi @jasonmccreary,

This was a side-effect of fixing another bug. The fix for this lies in one of flysystem's dependencies (league/mime-type-detection), which was shipped in version 1.11.0.
